tools/shared/src/observe-slot-text.ts

```diff
@@ -68,6 +69,12 @@ export function ObserveSlotText<T extends Constructor<ReactiveElement>>(
             const assignedNodes = [...this[assignedNodesList]].filter(
                 (node) => {
                     if ((node as HTMLElement).tagName) {
+                        if (exclude?.length) {
+                            const shouldExclude = exclude.some((selector) =>
+                                (node as HTMLElement).matches(selector)
+                            );
+                            return !shouldExclude;
+                        }
```

Comment:
Not a blocker, but I'm curious if defaulting `exclude` to `[]` and then simplifying the branching here would be equally-fast, eg:

```suggestion
                        return exclude.every((selector) => !(node as HTMLElement).matches(selector));
```

Reply:
I worried about a conditional argument with a default, but that's a good idea!